justinsoong iP

juzzztinsoong opened the pull request:
### Chat

Chat is a deranged and unconventional twist on personal assistant applications. Unlike its more sober counterparts, Chat boasts the riotous and unpredictable personality of [Twitch chat](https://www.twitch.tv/), making it a one-of-a-kind platform for managing your schedule and tasks. :skull:

> "Sadge" - Chat

What Chat does:
- Merges the practicality of a personal assistant with the delightful chaos of Twitch chat.
1. Offers a zany and motivating way to tackle daily responsibilities.
- [x] ~~Scares users away.~~ Ensures users never have a dull moment while staying organized.

Here's how to launch Chat from the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

juzzztinsoong committed:
renamed

juzzztinsoong committed:
ignore text files

juzzztinsoong committed:
echo

juzzztinsoong committed:
add, list

juzzztinsoong committed:
mark as done

juzzztinsoong committed:
partially implemented task subclasses

juzzztinsoong committed:
refactored task subclasses

juzzztinsoong committed:
refactored, added duke exceptions

juzzztinsoong committed:
added ability to delete tasks

juzzztinsoong committed:
implemented enum

juzzztinsoong committed:
testing

juzzztinsoong committed:
implemented file reader

juzzztinsoong committed:
added datetime info for tasks, partial implementation of datetimeparser

juzzztinsoong committed:
implemented file writer

juzzztinsoong committed:
implemented marking as done

juzzztinsoong committed:
saving implementation completed

juzzztinsoong committed:
Merge pull request #1 from juzzztinsoong/branch-Level-7

Branch level 7

juzzztinsoong committed:
Merge branch 'master' into branch-Level-8

juzzztinsoong committed:
merge + some time/date utility rewrites

juzzztinsoong committed:
implemented date time formatting

juzzztinsoong committed:
Merge pull request #2 from juzzztinsoong/branch-Level-8

Branch level 8

juzzztinsoong committed:
added OOP, packages already implemented

juzzztinsoong committed:
renamed packages

had to do something for this task

juzzztinsoong committed:
removed unnecessary code

juzzztinsoong committed:
Merge remote-tracking branch 'origin/add-gradle-support' into gradle-branch

juzzztinsoong committed:
got gradle working

juzzztinsoong committed:
Merge pull request #3 from juzzztinsoong/gradle-branch

Gradle branch

juzzztinsoong committed:
create test for tasklist and parser

juzzztinsoong committed:
rename files according to checkstyle, create jar file

juzzztinsoong committed:
reconfigure gradle build for jar

juzzztinsoong committed:
format based on coding standard

juzzztinsoong committed:
add javadocs to all required classes

juzzztinsoong committed:
implement find command

juzzztinsoong committed:
Merge pull request #4 from juzzztinsoong/branch-A-CodingStandard

format based on coding standard

juzzztinsoong committed:
Merge branch 'master' into branch-A-JavaDoc

juzzztinsoong committed:
Merge pull request #5 from juzzztinsoong/branch-A-JavaDoc

add javadocs to all required classes

juzzztinsoong committed:
remove todocommand

juzzztinsoong committed:
readd todocommand

file caps error

juzzztinsoong committed:
Merge branch 'master' into branch-Level-9

juzzztinsoong committed:
Merge pull request #6 from juzzztinsoong/branch-Level-9

implement find command

juzzztinsoong committed:
fix checkstyle violations

yuxunn commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,323 @@
+        String i2 = parsedInput[1].trim();
+
+        // Matches both halves of the input to the date.
+        if (i1.matches("[0-9]{2}.[0-9]{2}.[0-9]{4}")) {
```
the variables names i1 and i2 can be improved to make the code more self-explanatory and meaningful to increase code readability 😄

yuxunn commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,59 @@
+    @Override
+    public String toFileString() {
+        return String.format("E # %d # %s # %s %s # %s %s", (isDone ? 1 : 0), description, (
+            fromDate != null ? fromDate : ""), (fromTime != null ? fromTime : ""), (
```
the expressions here is abit complicated so maybe u can evaluate the expressions in steps first before the return statement. 😃

yuxunn commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,323 @@
+        if (i1.matches("[0-9]{2}.[0-9]{2}.[0-9]{2}")) {
+            time = LocalTime.parse(i1.substring(0, 2) + ":" + i1.substring(3, 5) + ":" + i1.substring(6, 8));
+        } else if (i2.matches("[0-9]{2}.[0-9]{2}.[0-9]{2}")) {
+            time = LocalTime.parse(i2.substring(0, 2) + ":" + i2.substring(3, 5) + ":" + i2.substring(6, 8));
```
this method has many if-else statements and is >30 lines of code for one method, perhaps the method can be shortened 😄

yuxunn reviewed (commented):
wow your code is amazing, the code quality is very high and you categorised the classes so neatly 😀 good job !!

juzzztinsoong committed:
implement gui

juzzztinsoong committed:
Merge pull request #7 from juzzztinsoong/branch-Level-10

implement gui

p-xp commented on src/main/java/chat/Chat.java:
```diff
@@ -0,0 +1,55 @@
+import chat.utils.Parser;
+import chat.utils.Storage;
+
+/**
+ * @author juzzztinsoong
+ */
```
It might be helpful to have a description of what this class does

p-xp commented on src/main/java/chat/Chat.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    /**
+     * Interacts with chat, sends command and receives output.
```
Perhaps you could specify sends command (to what) and receives output (from who / what)

p-xp commented on src/main/java/chat/commands/ByeCommand.java:
```diff
@@ -0,0 +1,24 @@
+
+    @Override
+    public String execute(TaskList tasklist, Storage storage) throws ChatException {
+        return "Aw goodbye.. ಠ_ಠ";
```
Cute!

p-xp commented on src/main/java/chat/commands/Command.java:
```diff
@@ -0,0 +1,43 @@
+import chat.tasks.TaskList;
+import chat.utils.Storage;
+
+/**
+ * @author juzzztinsoong
+ */
```
Maybe you could consider adding a class description? Noticed in several other classes as well.

p-xp commented on src/main/java/chat/controls/DialogBox.java:
```diff
@@ -0,0 +1,67 @@
+        displayPicture.setImage(img);
+    }
+
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
```
Clear description :)

p-xp commented on src/test/java/chat/tasks/TaskListTest.java:
```diff
@@ -0,0 +1,30 @@
+public class TaskListTest {
+
+    @Test
+    public void test1() {
```
Might be good to name tests using the format stated in the Java Coding Standards. As quoted below: 

Underscores may be used in test method names using the following three part format `featureUnderTest_testScenario_expectedBehavior()`.

p-xp reviewed (commented):
Overall LGTM! Just some nits. Really liked the way you handled abstraction btw.

juzzztinsoong committed:
fix jar compilation

Modify build.gradle and resource getter methods

This fixes jar compilation bugs.

Subsequently, can refactor resource getter methods to improve code
quality and streamline exception throwing process.

juzzztinsoong committed:
add assertions

Add a few assertions to launcher and parser.

Combine Launcher and Main class to streamline application flow.

juzzztinsoong committed:
Update Parser.java

juzzztinsoong committed:
Merge pull request #8 from juzzztinsoong/branch-A-Assertions

Branch a assertions

juzzztinsoong committed:
improve code quality

Refactor parser and tasklist to make happy path clearer and reduce code duplication.

Refactor tasks to reduce code complexity and increase readability.

Add javadoc description for every class.

To do:
Streamline GUI and application interaction.

juzzztinsoong committed:
Merge pull request #9 from juzzztinsoong/branch-A-CodeQuality

improve code quality

juzzztinsoong committed:
add sorting of tasks

Add ability to sort tasks by name, date and type.

Can look for more ways to sort tasks in future, but will require more parameters.

juzzztinsoong committed:
Merge pull request #10 from juzzztinsoong/branch-C-Sort

add sorting of tasks

juzzztinsoong committed:
improve GUI to look more like twitch chat

add app icon
add more chat images
add chat usernames

juzzztinsoong committed:
add chat background colors

juzzztinsoong committed:
add more memes

juzzztinsoong committed:
add user guide

juzzztinsoong committed:
Update README.md

juzzztinsoong committed:
Update README.md

juzzztinsoong committed:
fix for final release

fix storage issues
add more comprehensive tests

juzzztinsoong committed:
Merge branch 'master' of https://github.com/juzzztinsoong/ip

juzzztinsoong committed:
change javafx version to 11
